bin/waagent2.0

```diff
@@ -1569,6 +1569,7 @@ After=network.target
 After=sshd.service
 ConditionFileIsExecutable=/usr/sbin/waagent
 ConditionPathExists=/etc/waagent.conf
+ConditionVirtualization=microsoft
```

Comment:
You probably want `ConditionVirtualization=|microsoft` instead

> Condition checks can use a pipe symbol ("|") after the equals sign ("Condition…=|…"), which causes the condition to become a triggering condition. If at least one triggering condition is defined for a unit, then the unit will be started if at least one of the triggering conditions of the unit applies and all of the regular (i.e. non-triggering) conditions apply. If you prefix an argument with the pipe symbol and an exclamation mark, the pipe symbol must be passed first, the exclamation second. If any of these options is assigned the empty string, the list of conditions is reset completely, all previous condition settings (of any kind) will have no effect.

Reply:
@Conan-Kudo can you elaborate?  

Seems like it would make sense if we had an alternative trigger to run regardless of virtualization, e.g.:
```
ConditionVirtualization=|microsoft
ConditionPathExists=|/etc/run-waagent-even-if-not-hyperv
```

But given that no alternative trigger exists, is there a practical reason for making a lone triggering condition?

Reply:
I ask also because I recently made this change which is rolling down into RHEL: https://src.fedoraproject.org/rpms/hyperv-daemons/c/f4c66aab854882e15ce939196f076510b9c3729f?branch=rawhide :D

Reply:
I guess not? I saw that it was done that way for the VirtualBox guest agents, and I figure there was a good reason for that: 

* https://src.fedoraproject.org/rpms/virtualbox-guest-additions/blob/rawhide/f/vboxclient.service
* https://src.fedoraproject.org/rpms/virtualbox-guest-additions/blob/rawhide/f/vboxservice.service

Reply:
I updated it to use it as a triggering condition.  Thinking about it more it would make it easier for someone to extend the unit to run outside of hyper-v, if needed.